packages/pwa/app/components/_app-config/index.jsx

```diff
@@ -79,19 +84,30 @@ const AppConfig = ({children, locals = {}}) => {
     )
 }
 
-AppConfig.restore = (locals = {}) => {
-    // Parse the locale from the page url.
-    const locale = getLocale(locals) || DEFAULT_LOCALE
-    const currency = getPreferredCurrency(locale) || DEFAULT_CURRENCY
+// Question: Should the below 3 statics also have the `context` object set as their scope?
+// Remember that we'll have to text these function during rendering to see if they are
+// arrow functions or not and give a warning or error.
+AppConfig.restore = function() {}
+
+AppConfig.freeze = function() {
+    return undefined
+}
 
-    locals.api = new CommerceAPI({...apiConfig, locale, currency})
+AppConfig.freezeRequest = function(req) {
+    return {
+        url: req.url,
+        headers: {
+            'Accept-Language': req.headers['Accept-Language']
+        }
+    }
 }
 
-AppConfig.freeze = () => undefined
+AppConfig.extraGetPropsArgs = function() {
+    const locale = getLocale(this.originalRequest) || DEFAULT_LOCALE
```

Comment:
Chatted with Ben IRL but leave a record here for discussion:

It feels a little bit weird to access instance level property in a class static method. Developers who have experience with other OO programming languages, might find this confusing and hard to wrap head around. "What is `this`, why would a static method have access to an instance level property?"